obspy/core/util/base.py

```diff
@@ -60,6 +59,30 @@
 NATIVE_BYTEORDER = _sys_is_le and '<' or '>'
 
 
+# apply simple memoization cache on load_entry_points
+# this function cannot go in decorator.py due to circular import issues
+def _load_entry_point_decorator(func):
+    """
+    Decorate pkg_resources' load_entry_point function to cache outputs.
+    """
+    cache = {}
+    expected_keys = ('dist', 'group', 'name')
+
+    @functools.wraps(func)
+    def wrapper(*args, **kwargs):
+        # make a str key from the inputs to func for hashing in cache
+        input_dict = inspect.getcallargs(func, *args, **kwargs)
+        hash_str = '/'.join([input_dict[x] for x in expected_keys])
+        if hash_str not in cache:
+            cache[hash_str] = func(*args, **kwargs)
+        return cache[hash_str]
+
+    return wrapper
+
+
+load_entry_point = _load_entry_point_decorator(load_entry_point)
```

Comment:
Ah sorry - I did not read the PR carefully enough - yea then this is not used by ObsPy. It will really only set in `obspy.core.util.base.load_entry_point` so this would also have to be imported everywhere. It would of course be possible to monkey patch the `load_entry_point` function in `pkg_resources` but this is (i) dangerous for other packages and (ii) it would need to happen before it is imported everywhere else in ObsPy so we would introduce some hard-to-debug import order logic...

I'd personally favor to just import the modified `load_entry_point` everywhere in ObsPy - and maybe also just give it a different name and just code it into a function without the decorator as that would be easier to understand.

Can someone open a new PR to fix and also add a test for this?